detection_rules/rule_validators.py

```diff
@@ -182,14 +208,19 @@ def validate_stack_combos(self, data: QueryRuleData, meta: RuleMeta) -> None:
             eql_schema = ecs.KqlSchema2Eql(schema)
 
             # validate query against the beats and eql schema
-            self.validate_query_with_schema(data=data, schema=eql_schema, err_trailer=err_trailer,
-                                            beat_types=beat_types)
+            exc = self.validate_query_with_schema(data=data, schema=eql_schema, err_trailer=err_trailer,
+                                                  beat_types=beat_types)
+            if exc:
+                return exc
 
             if endgame_schema:
                 # validate query against the endgame schema
-                self.validate_query_with_schema(data=data, schema=endgame_schema, err_trailer=err_trailer)
+                output = self.validate_query_with_schema(data=data, schema=endgame_schema, err_trailer=err_trailer)
+                if output:
+                    raise ValueError(f"Endgame Schema error: {output}")
```

Comment:
Is this a ValueError if the message says schema error?

Reply:
Fixed in code :+1: